src/components/NcFilePicker/NcFilePicker.vue

```diff
@@ -0,0 +1,238 @@
+
+			<NcButton variant="tertiary"
+				@click="clearPicker">
+				Clear
+			</NcButton>
+		</div>
+
+		<h3>Selected files:</h3>
+		<ul>
+			<li v-for="file in selectedFiles" key="file.name">
+				{{ file.webkitRelativePath || file.name }}
+			</li>
+		</ul>
+	</div>
+</template>
+<script>
+	export default {
+		data() {
+			return {
+				selectedFiles: [],
+			}
+		},
+		methods: {
+			/**
+			 * This will clear the selected files from the picker.
+			 */
+			clearPicker() {
+				this.$refs.picker.reset()
+			},
+		},
+	}
+</script>
+<style scoped>
+.wrapper {
+	display: flex;
+	gap: 10px;
+}
+</style>
+```
+</docs>
+
+<script setup lang="ts">
+import type { Slot } from 'vue'
+
+import { computed, nextTick, useTemplateRef } from 'vue'
+import { t } from '../../l10n.js'
+import IconFolderUpload from 'vue-material-design-icons/FolderUpload.vue'
+import IconPlus from 'vue-material-design-icons/Plus.vue'
+import IconUpload from 'vue-material-design-icons/Upload.vue'
+import NcActionButton from '../NcActionButton/index.js'
+import NcActions from '../NcActions/index.js'
+
+const props = withDefaults(defineProps<{
+	/**
+	 * File types to accept
+	 */
+	accept?: string[]
+
+	/**
+	 * Allow picking a folder
+	 */
+	allowFolders?: boolean
+
+	/**
+	 * Disabled state of the picker
+	 */
+	disabled?: boolean
+
+	/**
+	 * If set then the label is only used for accessibility but not shown visually
+	 */
+	iconOnly?: boolean
+
+	/**
+	 * Label of the picker
+	 */
+	label?: string
+
+	/**
+	 * Can the user pick multiple files
+	 */
+	multiple?: boolean
+
+	/**
+	 * The variant of the button
+	 */
+	variant?: 'primary' | 'secondary' | 'tertiary'
+}>(), {
+	accept: undefined,
+	label: () => t('Pick files'),
+	variant: 'primary',
+})
+
+const emit = defineEmits<{
+	pick: [files: File[]]
+}>()
+
+defineSlots<{
+	/**
+	 * Custom NcAction* to be shown within the picker menu
+	 */
+	actions?: Slot
+
+	/**
+	 * Optional custom icon for the picker menu
+	 */
+	icon?: Slot
+
+	/**
+	 * Optional content to be shown in the picker.
+	 * This can be used e.g. for a progress bar or similar.
+	 */
+	default?: Slot
+}>()
+
+defineExpose({
+	reset,
+})
+
+const formElement = useTemplateRef('form')
+const inputElement = useTemplateRef('input')
+
+/**
+ * Check whether the current browser supports uploading directories
+ * Hint: This does not check if the current connection supports this, as some browsers require a secure context!
+ */
+const canUploadFolders = computed(() => {
+	return props.allowFolders && 'webkitdirectory' in document.createElement('input')
+})
+
+/**
+ * Trigger file picker
+ *
+ * @param uploadFolders - Whether to upload folders or files
+ */
+function triggerPickFiles(uploadFolders: boolean) {
+	// reset the form
+	reset()
+
+	// Setup directory picking if enabled
+	if (canUploadFolders.value) {
+		inputElement.value!.webkitdirectory = uploadFolders
+	}
+	// Trigger click on the input to open the file picker
+	nextTick(() => inputElement.value!.click())
+}
```

Comment:
Nitpick:

All the comments here just duiplicate what the code says (resets resets, click clicks), but doesn't explain the purpose (e.g. why mutate `webkitdirectory` and not bind in the template, why wait for `nextTick`).

```suggestion
function triggerPickFiles(uploadFolders: boolean) {
	// Without reset selecting the same file doesn't trigger the change event
	reset()

	// Vue does not support setting "webkitdirectory" via template <why>
	if (canUploadFolders.value) {
		inputElement.value!.webkitdirectory = uploadFolders
	}

	// Wait for <reason to wait>
	nextTick(() => inputElement.value!.click())
}
```